matminer/featurizers/composition/roost.py

```diff
@@ -0,0 +1,569 @@
+            "elem_gate": elem_gate,
+            "elem_msg": elem_msg,
+            "cry_heads": cry_heads,
+            "cry_gate": cry_gate,
+            "cry_msg": cry_msg,
+            "out_hidden": out_hidden,
+        }
+
+        model, criterion, optimizer, scheduler, normalizer = init_roost(
+            **self.setup_params,
+            **self.model_params,
+        )
+
+        self.model_name = model_name
+        self.run_id = run_id
+
+        self.model = model
+        self.criterion = criterion
+        self.optimizer = optimizer
+        self.scheduler = scheduler
+        self.normalizer = normalizer
+
+        self.model_name = model_name
+        self.run_id = run_id
+
+        self.elem_fea_len = elem_fea_len
+        self.elem_emb = elem_emb
+        self.task = task
+        self.epochs = epochs
+
+        self.fitted = False
+
+        # NOTE multiprocessing doesn't play nicely with pytorch, apparently you
+        # can set the multiprocessing start method to spawn but it's safer just
+        # to only use one job?
+        # set_start_method('spawn', force=True)
+        self.set_n_jobs(1)
+
+    def fit(self, X, y, checkpoint=False, log=False, verbose=False):
+
+        train_set = CompositionData(X, targets=y, task=self.task, emb=self.elem_emb)
+
+        # NOTE this is needed as we declare the model upfront without knowledge of the
+        # dataset.
+        if train_set.n_targets != self.model_params["n_targets"]:
+            raise ValueError("Number of targets for dataset does not match model")
+
+        train_generator = DataLoader(train_set, **self.data_params)
+
+        # if val_set is not None:
+        #     data_params.update({"batch_size": 16 * data_params["batch_size"]})
+        #     val_generator = DataLoader(self.val_set, **data_params)
+        # else:
+        #     val_generator = None
+
+        if self.model.task == "regression":
+            sample_target = torch.Tensor(train_set.targets)
+            self.normalizer.fit(sample_target)
+
+        # if (self.val_set is not None) and (self.model.best_val_score is None):
+        #     print("Getting Validation Baseline")
+        #     with torch.no_grad():
+        #         _, v_metrics = self.model.evaluate(
+        #             generator=self.val_generator,
+        #             criterion=self.criterion,
+        #             optimizer=None,
+        #             normalizer=self.normalizer,
+        #             action="val",
+        #             verbose=True,
+        #         )
+        #         if self.model.task == "regression":
+        #             val_score = v_metrics["MAE"]
+        #             print(f"Validation Baseline: MAE {val_score:.3f}\n")
+        #         elif self.model.task == "classification":
+        #             val_score = v_metrics["Acc"]
+        #             print(f"Validation Baseline: Acc {val_score:.3f}\n")
+        #         self.model.best_val_score = val_score
+
+        if checkpoint:
```

Comment:
Should `checkpoint` and `log` be paths? - up to you, but just a thought. That way people can specify where to keep their models and logs. The default can remain what it is right now.

Reply:
I think the default should be not to log or write files to peoples systems here unless they actively set the option but I think providing paths is a sensible change